lib/utils/index.js

```diff
@@ -11,7 +11,27 @@ function compose() {
   return funcs.reverse().reduce((a, b) => (...arg) => b(a(...arg)));
 }
 
+function getPropertyNames (obj) {
+  if (obj == null) return [];
+  const propertyNames = [];
+  // avoid to deep clone obj
+  let muteObj = Object.getPrototypeOf(obj);
+  do {
+    propertyNames.push.apply(propertyNames, Object.getOwnPropertyNames(muteObj));
+    muteObj = Object.getPrototypeOf(muteObj);
+  } while (muteObj);
+  muteObj = {};
+  // get own properties
+  propertyNames.push.apply(propertyNames, Object.getOwnPropertyNames(obj));
+  // get unique properties' names, filter __proto__
+  for (const name of propertyNames) {
+    muteObj[name] = 1;
+  }
```

Comment:
How about use `Set()` in the first place? Could be handier than filtering with object key.